### PR TITLE
Add some more to/refactor commands

### PR DESCRIPTION
**Pull request by technocake**
- Fixed circle-ci tests to run on py3.7 and tox.ini to use py3.7

Fixed some bugs in handling error messages 
Refactored text module to try to clearly show the difference between warning and error messages.

The distinction is a bit vague. I've been going by this rule:
- Warnings == when a user uses a command wrong, give him some helpful hints.
-  Errors == something outside goto / io / calling external commands etc. failed.

Added some more api-tests and added a unit test for the text module

### CODE REVIEW

**Comment by technocake** [comment]
I see now the text messages in the commands `cd`,  `copy` and `default` will fail. 
These are also coincidentally the commands without a api-test made yet..

File: goto/gotomagic/text/error.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+
```
Comment (Arxcis) [inline goto/gotomagic/text/error.py]: Great!  I wanted to create this file, but I tried to not do too many things at once. Glad you did it though 👍

File: goto/gotomagic/text/text.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# -*- code: utf-8 -*-
+"""
+    Text used by GOTO to do UX.
+"""
+from . import warning, error
+
+
+class GotoError:
```
Comment (Arxcis) [inline goto/gotomagic/text/text.py]: 👍

File: goto/gotomagic/text/text.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        # Making retrieval fail fast, if not present, stop and show the error.
+        # the missing message should then be made
+        text = error.messages.get(self.name, None)
+        if text is None:
+            raise Exception(
+                "Missing error message with name: {}".format(self.name))
+        return text.format(**self.kwargs)
+
+
+class GotoWarning:
```
Comment (Arxcis) [inline goto/gotomagic/text/text.py]: Great idea 👍

File: goto/tests/test_end_to_end.sh [hunk 4]
```diff
@@ -177,6 +177,59 @@ function test_07_goto_show {
     _failing_cmd_should_give_human_message "goto show $nonexisting_magicword"
 }
 
+function test_08_goto_rm {
```
Comment (Arxcis) [inline goto/tests/test_end_to_end.sh]: Can not get enough of these tests 👍 a

File: goto/tests/test_text.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+from unittest import TestCase
+from string import Formatter
+
+from ..gotomagic.text import GotoWarning, GotoError, error, warning
+
+
+def get_dict_with_format_args(message):
```
Comment (Arxcis) [inline goto/tests/test_text.py]: Can not get enough of these tests 👍 a

File: goto/the_real_goto.py [hunk 6]
```diff
@@ -9,20 +9,7 @@
...
-    show,\
-    copy,\
-    list,\
-    subl,\
-    vscode,\
-    intellij,\
-    open,\
-    cd,\
-    default
+from . import commands
```
Comment (Arxcis) [inline goto/the_real_goto.py]: Hmmm... I don't like removing the explicit imports 👎 but it is not a dealbreaker.

File: goto/the_real_goto.py [hunk 7]
```diff
@@ -34,9 +21,9 @@
 
 def main():
     if len(sys.argv) < 3:
-        output, _ = usage()
+        output, _ = commands.usage()
         print(output)
-        return
+        exit(0)
```
Comment (Arxcis) [inline goto/the_real_goto.py]: More explicit. I can dig it 👍

File: tox.ini [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # content of: tox.ini , put in same dir as setup.py
 [tox]
-envlist = py27,py36
+envlist = py27,py37
```
Comment (Arxcis) [inline tox.ini]: Horray!

**Pull request merged**